source/github-checks.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+const chalk = require('chalk');
+const ghGot = require('gh-got');
+const hostedGitInfo = require('hosted-git-info');
+const pTimeout = require('p-timeout');
+const pWaitFor = require('p-wait-for');
+const terminalLink = require('terminal-link');
+const git = require('./git-util');
+
+const THIRTY_SECONDS = 1000 * 30;
+const THIRTY_MINUTES = 1000 * 60 * 30;
+
+module.exports = async (task, pkg, options) => {
+	if (await git.hasUnpushedCommits()) {
+		if (options.preview) {
+			return task.skip('[Preview] Will not ensure checks have passed because there are unpushed commits.');
+		}
+
+		await git.push();
+	}
+
+	const latestCommit = await git.latestCommit();
+	const {title} = task;
+	const {user, project} = hostedGitInfo.fromUrl(pkg.repository.url);
+
+	const checkState = async () => {
+		try {
+			const response = await ghGot(`repos/${user}/${project}/commits/${latestCommit}/status`);
+			const {state, statuses} = response.body;
+
+			if (state === 'failure') {
+				task.title = title;
+
+				const failures = statuses.filter(status => status.state === 'error' || status.state === 'failure');
+				const contexts = failures.map(status => terminalLink(status.context, status.target_url));
+
+				if (failures.length === statuses.length) {
+					throw new Error(`All checks have failed: ${contexts.join(', ')}`);
+				} else {
+					throw new Error(`Some checks were not successful: ${contexts.join(', ')}`);
```

Comment:
Maybe

```suggestion
					throw new Error(`Some checks aren't passing: ${contexts.join(', ')}`);
```

Reply:
I’ve intentionally chosen the phrasing GitHub uses (see also 596a0a769f6e2232b5312af1a31ffc13b28ac81e):

<img width="397" alt="Screenshot" src="https://user-images.githubusercontent.com/652793/75753613-6d7b5e80-5d2b-11ea-8a80-740aa8c6dcb8.png">